rollup/fees/rollup_fee.go

```diff
@@ -48,59 +44,82 @@ type StateDB interface {
 	GetBalance(addr common.Address) *big.Int
 }
 
-// CalculateL1MsgFee computes the L1 portion of the fee given
-// a Message and a StateDB
-// Reference: https://github.com/ethereum-optimism/optimism/blob/develop/l2geth/rollup/fees/rollup_fee.go
-func CalculateL1MsgFee(msg Message, state StateDB) (*big.Int, error) {
-	tx := asTransaction(msg)
+func EstimateL1DataFeeForMessage(msg Message, baseFee, chainID *big.Int, signer types.Signer, state StateDB) (*big.Int, error) {
+	unsigned := asUnsignedTx(msg, baseFee, chainID)
+	tx, err := unsigned.WithSignature(signer, bytes.Repeat([]byte{0xff}, crypto.SignatureLength))
```

Comment:
Should we set `v` to 1 since the value could have impact on the `V` value in the EIP155 tx?